chronos/asyncloop.nim

```diff
@@ -665,8 +691,60 @@ elif unixPlatform:
       if not(isNil(adata.reader.function)):
         newEvents.incl(Event.Read)
     do:
-      raise newException(ValueError, "File descriptor not registered.")
-    loop.selector.updateHandle(int(fd), newEvents)
+      return err(OSErrorCode(osdefs.EBADF))
+
+    try:
+      loop.selector.updateHandle(int(fd), newEvents)
+    except CatchableError:
+      return err(osLastError())
+    ok()
+
+  proc register*(fd: AsyncFD) {.raises: [Defect, OSError].} =
+    ## Register file descriptor ``fd`` in thread's dispatcher.
+    let res = register2(fd)
+    if res.isErr(): raiseOSError(res.error())
```

Comment:
it's possible to create a `toException` function like so:
```nim
func toException(v: OsErrorCode): ref OsError = newOSError(v)
proc register*(fd: AsyncFD) {.raises: [Defect, OSError].} =
 register2(fd).tryGet()
```

it's not something that's widely used though, but maybe this is a good place